traitsui/qt4/instance_editor.py

```diff
@@ -218,9 +218,10 @@ def rebuild_items(self, event=None):
         if name >= 0:
             choice.setCurrentIndex(name)
         else:
-            # Otherwise, current value is no longer valid, try to discard it:
+            # Otherwise, current value is no longer valid, set combobox empty
             try:
                 self.value = None
+                choice.setCurrentIndex(name)
```

Comment:
So this is effectively `choice.setCurrentIndex(-1)`. Right?

Reply:
yes, perhaps just doing `choice.setCurrentIndex(-1)` is clearer / better.
I will update